satellite/audit/queue.go

```diff
@@ -0,0 +1,62 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package audit
+
+import (
+	"context"
+	"sync"
+
+	"storj.io/storj/pkg/storj"
+)
+
+// queue is a list of paths to audit, shared between the reservoir chore and audit workers.
+type queue struct {
+	cond   sync.Cond
+	queue  []storj.Path
+	closed chan struct{}
+}
+
+func newQueue() *queue {
+	return &queue{
+		cond:   *sync.NewCond(&sync.Mutex{}),
+		closed: make(chan struct{}),
+	}
+}
+
+// swap switches the backing queue slice with a new queue slice.
+func (queue *queue) swap(newQueue []storj.Path) {
+	queue.cond.L.Lock()
+	queue.queue = newQueue
+	// Notify workers that queue has been repopulated.
+	queue.cond.Broadcast()
+	queue.cond.L.Unlock()
+}
+
+// next gets the next item in the queue.
+func (queue *queue) next(ctx context.Context) (storj.Path, error) {
```

Comment:
Make it polling, since it eventually has to be.

Reply:
The polling happens usually outside of the queue, because it allows for callers to have more control and run other things while there's nothing in it.

See how repair queue is used.

Reply:
Updated to poll outside of the queue